src/collections/_documentation/development/sdk-dev/envelopes.md

```diff
@@ -0,0 +1,448 @@
+
+### Prerequisites
+
+These definitions apply to all parts of the Envelope data format:
+
+1. Newlines are defined as UNIX newlines, represented by `\n` and ASCII code 10.
+   If newlines are preceded with `\r`, this character is considered part of the
+   previous payload. This means that `\r\n` after is allowed to produce an
+   error.
+2. UUIDs are declared as 36 character strings and must contain
+   dashes:`"12c2d058-d584-4270-9aa2-eca08bf20986"`. It is recommended to use
+   UUID v4 in all cases.
+3. Envelopes do not offer a mechanism for compression. However, an entire
+   Envelope may be compressed or decompressed in an implementation defined way
+   by any component handling Envelopes. For example, [Ingestion](#ingestion)
+   allows compression via content encoding.
+
+### Headers
+
+Envelopes contain Headers in several places. Headers are JSON encoded objects
+(key-value mappings) that follow these rules:
+
+- Headers are always encoded in UTF-8 and must be valid JSON.
+- They must be declared in a single line, i.e. contain no newlines.
+- They must not be padded by any leading or trailing whitespace.
+- Whitespace within the JSON payload is permitted, although discouraged. That
+  is, JSON headers should be serialized in their most compact form without any
+  additional white space.
+- Unknown attributes are allowed and should be retained by all implementations.
+  However, attributes that are not part of this spec must not be actively
+  emitted by an implementation.
+- All known headers and their data types may be validated by an implementation.
+  If validation fails, the Envelope may be rejected as malformed.
+- Although not part of the Headers, Envelopes are always terminated by a newline
+  (`\n`) or the end of the file.
+- Empty headers `{}` are technically valid.
+
+Header-only Example:
+
+```json
+{"event_id":"12c2d058-d584-4270-9aa2-eca08bf20986"}
+```
+
+### Envelopes
+
+The full grammar for an Envelope is:
+
+```
+Envelope = Headers { "\n" Item } [ "\n" ] ;
+Item = Headers "\n" Payload ;
+Payload = { * } ;
+```
+
+- **Headers** are a single line containing a JSON object, as defined in the
+  *Headers* section. Attributes defined in the Envelope header scope the
+  contents of the Envelope and can be thought of applying to all items.
+
+- Based on the contents of the Envelope, certain attributes may be required. See
+  [Data Model](#data-model) for a specification of required attributes.
+- There can be an arbitrary number of **Items** separated by a newline. An
+  implementation should consume items until the file ends. Envelope may be
+  empty, terminating immediately after the headers.
+- Each Envelope should be terminated with a trailing newline. This newline is
+  optional. After the final newline, no whitespace is allowed.
+- The end of file (EOF) does not implicitly terminate an Envelope if more data
+  is expected.
+
+### Items
+
+Items carry actual information of an Envelope. Without items, an Envelope is
```

Comment:
Do we want to say "information"? (I'm not sure what our standard is). Perhaps "Items supply the data" or "Items supply the information for each Envelope" - I'm not quite sure "carry" is the correct verb here if we're getting across the idea that the Envelope is a kind of a wrapper, and the Items have the information used by Sentry, if I understand correctly.

Reply:
Yes, that's accurate. I'm going with "Items supply the data of an Envelope".